Create Simple_simulation

jwal1024 opened the pull request.

jwal1024 committed:
Create Simple_simulation

jwal1024 committed:
SPM weighting

Includes corrections to the SPM weightings.

pep8speaks commented:
Hello @jwal1024! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`jb/Child allowance.py`](https://github.com/UBICenter/child-allowance/blob/ee1f21846573d2925fde08740eea89102eebadf0/jb/Child allowance.py):

> [Line 4:80](https://github.com/UBICenter/child-allowance/blob/ee1f21846573d2925fde08740eea89102eebadf0/jb/Child allowance.py#L4): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)
> [Line 18:80](https://github.com/UBICenter/child-allowance/blob/ee1f21846573d2925fde08740eea89102eebadf0/jb/Child allowance.py#L18): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)
> [Line 22:29](https://github.com/UBICenter/child-allowance/blob/ee1f21846573d2925fde08740eea89102eebadf0/jb/Child allowance.py#L22): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 24:35](https://github.com/UBICenter/child-allowance/blob/ee1f21846573d2925fde08740eea89102eebadf0/jb/Child allowance.py#L24): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 25:37](https://github.com/UBICenter/child-allowance/blob/ee1f21846573d2925fde08740eea89102eebadf0/jb/Child allowance.py#L25): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 28:25](https://github.com/UBICenter/child-allowance/blob/ee1f21846573d2925fde08740eea89102eebadf0/jb/Child allowance.py#L28): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 29:80](https://github.com/UBICenter/child-allowance/blob/ee1f21846573d2925fde08740eea89102eebadf0/jb/Child allowance.py#L29): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (91 > 79 characters)
> [Line 31:80](https://github.com/UBICenter/child-allowance/blob/ee1f21846573d2925fde08740eea89102eebadf0/jb/Child allowance.py#L31): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (81 > 79 characters)
> [Line 37:80](https://github.com/UBICenter/child-allowance/blob/ee1f21846573d2925fde08740eea89102eebadf0/jb/Child allowance.py#L37): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (91 > 79 characters)
> [Line 46:37](https://github.com/UBICenter/child-allowance/blob/ee1f21846573d2925fde08740eea89102eebadf0/jb/Child allowance.py#L46): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 46:57](https://github.com/UBICenter/child-allowance/blob/ee1f21846573d2925fde08740eea89102eebadf0/jb/Child allowance.py#L46): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 51:80](https://github.com/UBICenter/child-allowance/blob/ee1f21846573d2925fde08740eea89102eebadf0/jb/Child allowance.py#L51): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (158 > 79 characters)
> [Line 51:159](https://github.com/UBICenter/child-allowance/blob/ee1f21846573d2925fde08740eea89102eebadf0/jb/Child allowance.py#L51): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 53:33](https://github.com/UBICenter/child-allowance/blob/ee1f21846573d2925fde08740eea89102eebadf0/jb/Child allowance.py#L53): [E712](https://duckduckgo.com/?q=pep8%20E712) comparison to True should be 'if cond is True:' or 'if cond:'
> [Line 53:80](https://github.com/UBICenter/child-allowance/blob/ee1f21846573d2925fde08740eea89102eebadf0/jb/Child allowance.py#L53): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (96 > 79 characters)
> [Line 54:33](https://github.com/UBICenter/child-allowance/blob/ee1f21846573d2925fde08740eea89102eebadf0/jb/Child allowance.py#L54): [E712](https://duckduckgo.com/?q=pep8%20E712) comparison to False should be 'if cond is False:' or 'if not cond:'
> [Line 54:80](https://github.com/UBICenter/child-allowance/blob/ee1f21846573d2925fde08740eea89102eebadf0/jb/Child allowance.py#L54): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (93 > 79 characters)
> [Line 56:44](https://github.com/UBICenter/child-allowance/blob/ee1f21846573d2925fde08740eea89102eebadf0/jb/Child allowance.py#L56): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 57:38](https://github.com/UBICenter/child-allowance/blob/ee1f21846573d2925fde08740eea89102eebadf0/jb/Child allowance.py#L57): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 59:31](https://github.com/UBICenter/child-allowance/blob/ee1f21846573d2925fde08740eea89102eebadf0/jb/Child allowance.py#L59): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 62:80](https://github.com/UBICenter/child-allowance/blob/ee1f21846573d2925fde08740eea89102eebadf0/jb/Child allowance.py#L62): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)
> [Line 62:80](https://github.com/UBICenter/child-allowance/blob/ee1f21846573d2925fde08740eea89102eebadf0/jb/Child allowance.py#L62): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)
> [Line 62:85](https://github.com/UBICenter/child-allowance/blob/ee1f21846573d2925fde08740eea89102eebadf0/jb/Child allowance.py#L62): [W292](https://duckduckgo.com/?q=pep8%20W292) no newline at end of file
> [Line 62:85](https://github.com/UBICenter/child-allowance/blob/ee1f21846573d2925fde08740eea89102eebadf0/jb/Child allowance.py#L62): [W292](https://duckduckgo.com/?q=pep8%20W292) no newline at end of file
> [Line 62:85](https://github.com/UBICenter/child-allowance/blob/ee1f21846573d2925fde08740eea89102eebadf0/jb/Child allowance.py#L62): [W292](https://duckduckgo.com/?q=pep8%20W292) no newline at end of file

MaxGhenis commented:
Re pep8, I'd recommend using the `black` code formatter VSCode extension, that way you can fix all of them in one go (also change the line length black config to 79): https://code.visualstudio.com/docs/python/editing#_formatting

jwal1024 committed:
Formatting using black 79

jwal1024 committed:
poverty rate calc; Please delete old file for rename

Compute poverty rate by policy + rename file + clean up notes. Please delete old file for rename

jwal1024 committed:
Delete Simple_simulation

Replaced by child_allowance.py

jwal1024 committed:
Delete Child allowance.py

Replaced by child_allowance.py

jwal1024 committed:
% changes in poverty rate

include, % changes in poverty rate, formatting corrections. May be an outstanding issue with code lines for weighted summs - need to rerun it on my end will have a further commit within the next few hours.

jwal1024 committed:
Update child_allowance.py

Corrections for poverty rate calculation code + indexing sim flags from 1-3 rather than 0-2 for consistency.
Still an indexing issue in lines 118, 120, 124.

jwal1024 committed:
Test inclusion IPUMS data

Test inclusion IPUMS data

jwal1024 committed:
Update child_allowance.py

Changes made with Max to calculate heterogenous poverty effects. Path updated in commit later tonight.

jwal1024 committed:
Replace dataset, work on race cats.

Replace dataset to include HISPAN and remove monthly dataset. Define race cats.

jwal1024 committed:
Race categories, prelim results

jwal1024 committed:
Gini coefficients, child poverty, states.

Code put in for gini coefficients, child poverty, state labels. Dataframes renamed for clarity. Notes at beginning and end clarified.  May be one more commit to meet PEP8 code guidelines.

jwal1024 committed:
Outstanding: child poverty rate

Fixed child_6 definitions.

MaxGhenis commented on jb/child_allowance.py:
```diff
@@ -0,0 +1,236 @@
+)
+
+# Define data collected at the SPM unit level
+spmu_cols = [
```
Make `SPMU_COLS` since it's a constant

jwal1024 replied:
As in just capitalize all instances of SPMU_COLS?

MaxGhenis commented on jb/child_allowance.py:
```diff
@@ -0,0 +1,236 @@
+    "Black",
+    inplace=True,
+)
+person.race_hispan.mask(
```
Can delete this thru line 85, since it's the default

MaxGhenis commented on jb/child_allowance.py:
```diff
@@ -0,0 +1,236 @@
+person["person"] = 1
+
+# Redefine race categories
+person.race.mask(
```
Could you make this a new column instead, e.g. `person.race_group`? Often safer to avoid overloading original column names with new concepts. Same with `hispan` below.

MaxGhenis commented on jb/child_allowance.py:
```diff
@@ -0,0 +1,236 @@
+)
+person.race.mask(person.race == 100, "White", inplace=True)
+person.race.mask(person.race == 200, "Black", inplace=True)
+person.race.mask(person.race == 999, "Race unknown", inplace=True)
```
Unnecessary, can delete

MaxGhenis commented on jb/child_allowance.py:
```diff
@@ -0,0 +1,236 @@
+person.race_hispan.mask(person.hispan == "Hispanic", "Hispanic", inplace=True)
+
+# Relabel sex categories
+person["sex"].mask(person["sex"] == 1, "Male", inplace=True)
```
Can replace this and the next with `person["female"] = person.sex == 2`

MaxGhenis commented on jb/child_allowance.py:
```diff
@@ -0,0 +1,236 @@
+# Create a copy of the raw dataset and make column names non-capitalized
+# for readability
+person = raw.copy(deep=True)
+person = person[~person.ASECWT.isnull()]
```
Do any person records have null weight?

MaxGhenis commented on jb/child_allowance.py:
```diff
@@ -0,0 +1,236 @@
+    person.groupby(spmu_cols)[
+        ["child_6", "infant", "toddler", "preschool", "person"]
+    ].sum()
+).reset_index()
```
Could be clearer to rename columns after this, e.g. starting from 110:
```
SPMU_AGG_COLS = ["child_6", "infant", "toddler", "preschool", "person"]
spmu = person.groupby(SPMU_COLS)[SPMU_AGG_COLS].sum()  # Wait to reset_index().
spmu.columns = ["spmu_" + i for i in SPMU_AGG_COLS]
spmu.reset_index(inplace=True)
```

MaxGhenis commented on jb/child_allowance.py:
```diff
@@ -0,0 +1,236 @@
+
+# Generate simulation-level flags to separate datasets,
+# 0 = base case, 1 = cost replacement design, 2 = flat transfer
+spmu["sim_flag"] = "baseline"
```
something like "scenario" might be clearer

MaxGhenis commented on jb/child_allowance.py:
```diff
@@ -0,0 +1,236 @@
+)
+
+# Consider sex, race, state heterogeneity
+poverty_rate = (
```
Can simplify these statements with a function, e.g.:
```
def pov(groupby, filter=True):
    return person_sim[filter].groupby(groupby).apply(lambda x: mdf.weighted_mean(x, "poverty_flag", "asecwt")).reset_index()
```
(with proper line length)

then you'd have:
```
poverty_rate = pov("sim_flag")
poverty_rate_sex = pov(["sim_flag", "sex"])
...
```
and also, via the optional `filter` arg:
```
poverty_u6 = pov("sim_flag", person_sim.child_6)
```

MaxGhenis commented on jb/data/cps_00003.csv.gzZone.Identifier:
```diff
@@ -0,0 +1,4 @@
+[ZoneTransfer]
```
remove this file

MaxGhenis commented on jb/t-bc890c1b-0f3d-4562-a58d-dd8c1f812f9a.ipynb:
```diff
@@ -0,0 +1,6 @@
+{
```
remove this? and other 3 weirdly named ipynb files

MaxGhenis commented on jb/.vscode/settings.json:
```diff
@@ -0,0 +1,3 @@
+{
```
remove this

MaxGhenis commented on jb/child_allowance.py:
```diff
@@ -0,0 +1,236 @@
+
+# Read in census data and specify columns for use
+raw = pd.read_csv(
+    "https://github.com/UBICenter/child-allowance/blob/master/jb/data/cps_00003.csv.gz",
```
You can load this locally right? i.e. `pd.read_csv("data/cps_00003.csv.gz")`

jwal1024 committed:
Delete settings.json

jwal1024 committed:
Delete t-bc890c1b-0f3d-4562-a58d-dd8c1f812f9a.ipynb

jwal1024 committed:
Delete t-bda112fb-69fc-4587-a9f8-043a9ca2a9e1.ipynb

jwal1024 committed:
Delete t-dc1856b5-1a9c-402a-a8ec-c7358321609f.ipynb

jwal1024 committed:
Delete cps_00003.csv.gzZone.Identifier

jwal1024 committed:
Delete make_decile_data.py

jwal1024 committed:
First round of Max's comments

jwal1024 committed:
Update child_allowance.py

More of Max's changes.

jwal1024 committed:
Merge branch 'patch-1' of https://github.com/jwal1024/childallowance into patch-1

jwal1024 committed:
Race/hispan definitions

Final corrections to race/hispan definitions to push unknowns into category other.

MaxGhenis commented:
Thanks @jwal1024 for this PR to simulate the distributional effects of (a) a transfer equal to a SPM unit's current childcare expenses, (b) a universal child allowance for children under 6.
